src/matcha_ml/cli/cli.py

```diff
@@ -119,11 +122,17 @@ def destroy(
 ) -> None:
     """Destroy the provisioned cloud resources."""
     remote_state_manager = RemoteStateManager()
-
-    destroy_resources()
-
-    if full:
-        remote_state_manager.deprovision_remote_state()
+    with Spinner("Preparing to destroy..."):
```

Comment:
This seems to be using UI inside `cli.py`. 

Note: In addition to this spinner, there will be either 1 or 2 spinners (depending on full or not) implemented inside `_destroy_terraform` [here](https://github.com/fuzzylabs/matcha/blob/develop/src/matcha_ml/runners/base_runner.py#L158).

Reply:
The first spinner will keep running even when the rest of spinners are completed. The UI would be

- The first spinner with `Preparing to destroy starts running`
- We get few messages about destroying
- Second spinner about destory starts running and completes
- We get few more messages
- Thrid spinner about destroying matcha resources starts running and completes

All this spinners are completed and first spinner is still running (at the start of the UI), so I don't think it would any good here. Instead there should be spinners for `download` and `upload` functions inside `use_remote_state`.

I encourage you to test to actually visualize this.